python/ray/serve/_private/proxy_router.py

```diff
@@ -46,11 +59,15 @@ def __init__(
         self.handles: Dict[DeploymentID, DeploymentHandle] = dict()
         # Map of application name to is_cross_language.
         self.app_to_is_cross_language: Dict[ApplicationName, bool] = dict()
+        # Flipped to `True` once the route table has been updated at least once.
+        # The proxy router is not ready for traffic until the route table is populated
+        self._route_table_populated = False
 
     def update_routes(self, endpoints: Dict[DeploymentID, EndpointInfo]) -> None:
         logger.info(
             f"Got updated endpoints: {endpoints}.", extra={"log_to_stderr": False}
         )
+        self._route_table_populated = True
```

Comment:
should this also require that the size of the route_table is > 0?

Reply:
yeah that makes sense, changed so that it is only flipped on if a nonzero list of routes were received.